apps/native-component-list/components/Player.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import { ScrollView, Slider, StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { Audio } from 'expo';
+import { Ionicons } from '@expo/vector-icons';
+
+import Colors from '../constants/Colors';
+
+export default class Player extends React.Component {
+  state = {
+    userIsDraggingSlider: false,
+    isLoaded: false,
+    isLooping: false,
+    errorMessage: null,
+    positionMillis: 0,
+    durationMillis: 0,
+  };
+
+  componentDidMount() {
+    this._loadSoundAsync();
+  }
+
+  _loadSoundAsync = async () => {
+    const soundObject = new Audio.Sound();
+    try {
+      await soundObject.loadAsync(this.props.source, { progressUpdateIntervalMillis: 100 });
+      soundObject.setOnPlaybackStatusUpdate(this._updateStateToStatus);
+      const status = await soundObject.getStatusAsync();
+      this._updateStateToStatus(status);
+      this._sound = soundObject;
+    } catch (error) {
+      this.setState({ errorMessage: error.message });
+    }
+  };
+
+  _updateStateToStatus = status => this.setState(status);
+
+  _play = () => this._sound.playAsync();
+  _pause = () => this._sound.pauseAsync();
+  _playFromPosition = position =>
+    this._sound
+      .setPositionAsync(position)
+      .then(() => this.setState({ userIsDraggingSlider: false }));
+  _toggleLooping = () => this._sound.setIsLoopingAsync(!this.state.isLooping);
+  _toggleSlowRate = () =>
+    this._sound.setRateAsync(this.state.rate < 1 ? 1 : 0.5, this.state.shouldCorrectPitch);
+  _toggleFastRate = () =>
+    this._sound.setRateAsync(this.state.rate > 1 ? 1 : 2, this.state.shouldCorrectPitch);
+  _toggleShouldCorrectPitch = () =>
+    this._sound.setRateAsync(this.state.rate, !this.state.shouldCorrectPitch);
```

Comment:
Can you prettify this whole block of methods? Even line gaps between them would make this block look much better 😛